### ISSUE TITLE
Fatal Exception: NSInternalInconsistencyException

### ISSUE THREAD

**Issue by Shain-Dholakiya** [DataDog/dd-sdk-flutter#390]
Thanks for taking the time for reporting an issue!

**Describe what happened**
Include any error message or stack trace if available.

partial apply for closure https://github.com/DataDog/dd-sdk-ios/pull/1 in SwiftDatadogSdkPlugin.handle(_:result:)
NSInternalInconsistencyException - Sending a message before the FlutterEngine has been run.

Fatal Exception: NSInternalInconsistencyException
0 CoreFoundation 0xa248 __exceptionPreprocess
1 libobjc.A.dylib 0x17a68 objc_exception_throw
2 Foundation 0x54681c userInfoForFileAndLine
3 Flutter 0x12eb4 (Missing UUID 66de630a2b913f7d9856a3b1841da964)
4 Runner 0x1100cc partial apply for closure https://github.com/DataDog/dd-sdk-ios/pull/1 in SwiftDatadogSdkPlugin.handle(:result:) ()
5 Datadog 0x1522a4 (Missing UUID b65fc4e76a9237bd86862888604e503c)
6 Datadog 0xd1d7c (Missing UUID b65fc4e76a9237bd86862888604e503c)
7 Datadog 0x141e38 (Missing UUID b65fc4e76a9237bd86862888604e503c)
8 Datadog 0x6dd50 (Missing UUID b65fc4e76a9237bd86862888604e503c)
9 Datadog 0x6dd70 (Missing UUID b65fc4e76a9237bd86862888604e503c)
10 libdispatch.dylib 0x3fdc _dispatch_client_callout
11 libdispatch.dylib 0x13574 _dispatch_lane_barrier_sync_invoke_and_complete
12 Datadog 0x141d0c (Missing UUID b65fc4e76a9237bd86862888604e503c)
13 Datadog 0x145920 (Missing UUID b65fc4e76a9237bd86862888604e503c)
14 Datadog 0x678a0 (Missing UUID b65fc4e76a9237bd86862888604e503c)
15 Datadog 0x5f6a0 (Missing UUID b65fc4e76a9237bd86862888604e503c)
16 Datadog 0x13d120 (Missing UUID b65fc4e76a9237bd86862888604e503c)
17 libdispatch.dylib 0x24b4 _dispatch_call_block_and_release
18 libdispatch.dylib 0x3fdc _dispatch_client_callout
19 libdispatch.dylib 0xb694 _dispatch_lane_serial_drain
20 libdispatch.dylib 0xc1e0 _dispatch_lane_invoke
21 libdispatch.dylib 0x16e10 _dispatch_workloop_worker_thread
22 libsystem_pthread.dylib 0xdf8 _pthread_wqthread
23 libsystem_pthread.dylib 0xb98 start_wqthread

**Steps to reproduce the issue:**

**Describe what you expected:**

**Additional context**
 - Dart/Flutter version: 3.3.3
 - Android/iOS OS version: 16
 - Device Model: iPhone 11, iPhone 14 Pro, iPhone 13 Pro Max
 - Datadog SDK version: 1.17.0
 - Versions of any other relevant dependencies:

**Comment by fuzzybinary** [comment]
Hi @Shain-Dholakiya,

Is there any more information you can give us about this crash? Do you have the dSYMs that were generated with the build, or have any idea what causes it?

The exception:
```
NSInternalInconsistencyException - Sending a message before the FlutterEngine has been run.
```

Is it possible this is duplicate of the crash fixed in 1.2.2? https://github.com/DataDog/dd-sdk-flutter/issues/341

**Comment by Shain-Dholakiya** [comment]
@fuzzybinary let me try updating the version and then seeing if the crashes are still happening on production. If it still is, I will try to get you the generated dSYMs

**Comment by fuzzybinary** [comment]
@Shain-Dholakiya I'd appreciate it.

Remember that you are reporting errors and crashes to Datadog, can upload your dSYM and symbol files to Datadog and we'll symbolicate crashes for you:

https://docs.datadoghq.com/real_user_monitoring/error_tracking/flutter/#overview

**Comment by followthemoney1** [comment]
-

**Comment by fuzzybinary** [comment]
@followthemoney1 What version of the SDK are you on? This appears to be the crash on exit that I believe was fixed in 1.2.2. If you're on 1.2.2+ let me know and I'll re-open the investigation.

**Comment by followthemoney1** [comment]
> @followthemoney1 What version of the SDK are you on? This appears to be the crash on exit that I believe was fixed in 1.2.2. If you're on 1.2.2+ let me know and I'll re-open the investigation.

I've updated it to 1.3.0, and I let you know if this happens with the new version, needs some time. Cuz this problem is appears on `flutterMode: release`

**Comment by fuzzybinary** [comment]
@muccy Thanks, I'll reopen the investigation. 

It's incredibly timing specific so hard to pin down why Flutter isn't properly calling to shutdown the method channel, but I'll try to see if I can create a workaround.

**Comment by fuzzybinary** [comment]
@followthemoney1 Actually, your crash and @Shain-Dholakiya 's are different. Yours is an instance of #341 which should be fixed in 1.2.2 (please let me know if not.

@muccy Can you verify if what you're seeing are crashes with `NSInternalInconsistencyException`, or with the `SIGABRT` from `NSAssertionHandler`? In both cases, do you have any stack traces, and can you let me know what Flutter version you're on?

**Comment by amaurySaugrain** [comment]
Hi! I am also seeing the `Fatal Exception: NSInternalInconsistencyException Sending a message before the FlutterEngine has been run.`. I am running version 1.3.2

```
0
CoreFoundation
__exceptionPreprocess
1
libobjc.A.dylib
objc_exception_throw
2
Foundation
_userInfoForFileAndLine
3
Flutter
(Missing UUID be105c8b7c053a8ba51f83e291732357)
4
Runner
<compiler-generated>
partial apply for closure #1 in SwiftDatadogSdkPlugin.handle(_:result:)
5
Datadog
<compiler-generated>
specialized RUMSessionScope.process(command:context:writer:)
6
Datadog
<compiler-generated>
RUMApplicationScope.process(command:context:writer:)
7
Datadog
RUMMonitor.swift - Line 623
closure #1 in closure #1 in RUMMonitor.process(command:) + 623
8
Datadog
<compiler-generated>
thunk for @callee_guaranteed () -> ()
9
Datadog
<compiler-generated>
thunk for @escaping @callee_guaranteed () -> ()
10
libdispatch.dylib
_dispatch_client_callout
11
libdispatch.dylib
_dispatch_lane_barrier_sync_invoke_and_complete
12
Datadog
<compiler-generated>
closure #1 in RUMMonitor.process(command:)
13
Datadog
<compiler-generated>
partial apply for closure #1 in RUMMonitor.process(command:)
14
Datadog
DatadogCore.swift - Line 486
closure #1 in DatadogCoreFeatureScope.eventWriteContext(bypassConsent:forceNewBatch:_:) + 486
15
Datadog
DatadogContextProvider.swift - Line 105
closure #1 in DatadogContextProvider.read(block:) + 105
16
Datadog
<compiler-generated>
thunk for @escaping @callee_guaranteed () -> ()
17
libdispatch.dylib
_dispatch_call_block_and_release
18
libdispatch.dylib
_dispatch_client_callout
19
libdispatch.dylib
_dispatch_lane_serial_drain
20
libdispatch.dylib
_dispatch_lane_invoke
21
libdispatch.dylib
_dispatch_workloop_worker_thread
22
libsystem_pthread.dylib
_pthread_wqthread
23
libsystem_pthread.dylib
start_wqthread
```

**Comment by fuzzybinary** [comment]
@amaurySaugrain What Flutter version are you on and do you have the dump from the other threads?

**Comment by fuzzybinary** [comment]
🤔 Okay, re-evaluating this does seem to be the same as #341. I'm going to re-open investigation and I'll get back to everyone.

**Comment by fuzzybinary** [comment]
PR is up for a fix and will go out in 1.4.0 later this week. I will also try to get this fix into the 1.3.x line as the Datadog Plugin 1.4+ will require Flutter 3+.
